### ISSUE TITLE
Cloudflare stops the API access

### ISSUE THREAD

**Issue by altmind** [mpolden/echoip#166]
While trying to access the `curl ifconfig.co/` I'm getting cloudlfare challenge that is impossible to do in curl. The "How do I get this programmatically?" section does not work.
```html
<!DOCTYPE html>
<html lang="en-US">
<head>
    <title>Just a moment...</title>
...
        <h2 class="h2" id="challenge-running">
            Checking if the site connection is secure
        </h2>
```

Remove the cloudflare, of this website is not usable for its intended purpose.

**Comment by altmind** [comment]
Sourcing from 35.79.238.0/24, but im not asking to whitelist this subnet. Cloudlfare defeats the purpose of API access, cloudflare needs to be removed?

**Comment by mfld-pub** [comment]
Cloudflare enterprise can make rules to bypass bot prevention with granularity. 

Free, pro and business can only turn it on or off. 

Uptime-Kuma project has the same issue. 

Cloudflare itself doesn't break this. We have our own instance of echo IP behind cf and query with curl. The bot protection creating false positives is the issue.

**Comment by altmind** [comment]
well, the IP address that had <10/accesses to ifconfig.co in the past 90days is getting blocked by cloudflare.

cloudflare is falsely blocking non-abusing IP address.

**Comment by mfld-pub** [comment]
> well, the IP address that had <10/accesses to ifconfig.co in the past 90days is getting blocked by cloudflare.
> 
> cloudflare is falsely blocking non-abusing IP address.

Yes, they are. The only remedy is to go on an expensive plan or disable bot fighting option in CF for the domain. IIRC it is on by default. 

They have a way to get certain clients or apps whitelisted but this is not possible for python -requests or curl as that's exactly what a bot would use.

**Comment by altmind** [comment]
im asking to disable cloudflare for the main website

**Comment by mfld-pub** [comment]
I don't think this will be done. It would expose the origin server and lose ddos protection. The compromise is to toggle "bot fighting mode" off.

**Comment by altmind** [comment]
well, to bad i cannot use the website because of false positives

**Comment by fiskhest** [comment]
@mpolden would you consider toggling the cloudflare `bot fighting mode` off? See above comments from @mfld-pub

**Comment by mpolden** [comment]
@dhrp is in charge of hosting now. Maybe we can try turning off the bot-fighting mode?

**Comment by giladreich** [comment]
I'm experiencing the same issue, especially when running in CI/CD environment using GitHub's runners or AWS EC2 instances. @mpolden @dhrp, would you be so kind to disable the bot-fighting mode? Thank you!

**Comment by hydrargyrum** [comment]
Yes, the service is worthless as is, but fortunately there are alternatives that actually work:

```
curl https://api.ipify.org/
curl -6 https://api64.ipify.org/
```

**Comment by giladreich** [comment]
@mpolden or any maintainer? ping.

**Comment by mfld-pub** [comment]
Simple solution for me was to self host the project on a domain of my choosing with LB/WAF settings of my choosing. Runs well and requires very very little resources.

Perhaps y'all could consider doing that.

**Comment by giladreich** [comment]
> Simple solution for me was to self host the project on a domain of my choosing with LB/WAF settings of my choosing. Runs well and requires very very little resources.
> 
> Perhaps y'all could consider doing that.

That option is kind of obvious, but then again at the cost of maintenance. Mind sharing your process of self-hosting it? How much resources you gave the instance? Did you implement a load balancer? What about updating the Geo location databases, did you implement automation for it to auto-update? AFAIK also for getting the Geo location databases it requires creating an account and such.

**Comment by ifconfigla** [comment]
For anyone finding the same problem I made the host of ifconfig.la which is cloudflare-free.

**Comment by hydrargyrum** [comment]
`ifconfig.la` doesn't support ipv6

**Comment by ifconfigla** [comment]
@hydrargyrum It is intentional since we could not check tor exit nodes against IPv6 exit nodes. If some amount of donation is received will host another instance under a new domain that will support both IPv4 and IPv6.

**Comment by dxdxdt** [comment]
Yes. Just another useless tool made by incompetent devs. I wonder what it takes to get the site off the search engines.
